src/Sarif.Sarifer/BackgroundAnalysisService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    /// <summary>
+    /// Performs static analysis in the background.
+    /// </summary>
+    // TODO: Reanalyze when buffer changes.
+    // TODO: Remove error list items when buffer closes.
+    [Export(typeof(IBackgroundAnalysisService))]
+    internal class BackgroundAnalysisService : IBackgroundAnalysisService
+    {
+#pragma warning disable CS0649 // Filled in by MEF
+#pragma warning disable IDE0044 // Assigned by MEF
+        [ImportMany]
+        private IEnumerable<IBackgroundAnalyzer> analyzers;
+#pragma warning restore IDE0044
+#pragma warning restore CS0649
+
+        /// <inheritdoc/>
+        public void StartAnalysis(string path, string text)
+        {
+            if (this.analyzers.Any() == true)
+            {
+                foreach (IBackgroundAnalyzer analyzer in this.analyzers)
```

Comment:
>this.analyzers [](start = 57, length = 14)

If this.analyzers cannot be null, then the any isn't needed at all right?

Reply:
Right. The worst it can be is empty, so you're right, the check isn't needed.

---
In reply to: [519917807](https://github.com/microsoft/sarif-visualstudio-extension/pull/260#discussion_r519917807) [](ancestors = 519917807)